From 0c02f4b7ae416d5f485ba09cfaf9bee0a93dcc90 Mon Sep 17 00:00:00 2001
From: Akshay Gupta <Akshay.Gupta@amd.com>
Date: Mon, 10 Apr 2023 04:33:55 -0400
Subject: i3c: move i3c_bus_type definition to linux header

At present, "struct bus_type i3c_bus_type" is defined in driver/i3c/internals.h.
Moving it to a standard include path, this way i3c client devices
outside driver/i3c/ can utilize it.

Signed-off-by: Akshay Gupta <Akshay.Gupta@amd.com>
Signed-off-by: Naveen Krishna Chatradhi <naveenkrishna.chatradhi@amd.com>
---
 drivers/i3c/internals.h    | 1 -
 include/linux/i3c/device.h | 1 +
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/i3c/internals.h b/drivers/i3c/internals.h
index a6deedf5ce06..3414b6627150 100644
--- a/drivers/i3c/internals.h
+++ b/drivers/i3c/internals.h
@@ -10,7 +10,6 @@

 #include <linux/i3c/master.h>

-extern struct bus_type i3c_bus_type;
 extern const struct device_type i3c_masterdev_type;

 void i3c_bus_normaluse_lock(struct i3c_bus *bus);
diff --git a/include/linux/i3c/device.h b/include/linux/i3c/device.h
index ad18deff6574..b915cbeee184 100644
--- a/include/linux/i3c/device.h
+++ b/include/linux/i3c/device.h
@@ -15,6 +15,7 @@
 #include <linux/mod_devicetable.h>
 #include <linux/module.h>

+extern struct bus_type i3c_bus_type;
 /**
  * enum i3c_error_code - I3C error codes
  *
--
2.27.0

